File: backend/Services/Parsers/LewdleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class LewdleParser : BasicScoreResultParser
    {
        private ILogger<LewdleParser> _logger;

        public LewdleParser(ILogger<LewdleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override DateTime ActiveAfter => DateTime.MinValue;
        public override string Category => "Word";
        public override string GameName => "Lewdle";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"{GameName}.*?(?<{ScoreGroup}>[\\dxX])/\\d");
        protected override string? ExtraContent => "lewdlegame.com";
        public override string Url => "https://www.lewdlegame.com";
    }
}

File: lambdas/src/parsers/lewdle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class LewdleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Lewdle';
  readonly helpText = null;
  protected readonly extraContent = 'lewdlegame.com';
  readonly url = 'https://www.lewdlegame.com';
  
  // Matches "Lewdle 🍆💦 122 x/6" or "Lewdle 🍆💦 100 2/6"
  protected readonly parser = new RegExp(
    `${this.gameName}.*?(?<${BasicScoreResultParser.SCORE_GROUP}>[\\dxX])/\\d`
  );
}
